### PR TITLE
feat: upgrade to react 18 and chakra v2

### PR DESCRIPTION
**Pull request by karrui**
This PR upgrades React from React 17 to React 18, as well as ChakraUI v1 to v2. This is necessary due to ChakraUI V2 only being compatible with React 18. 

In addition, a bunch of packages are upgraded too to allow for compatibility with React 18.

list of changed deps required to make the app work lol:
* "@chakra-ui/react": "^1.8.6"  -> ^2.8.2
* "@floating-ui/react-dom-interactions": "^0.9.3" (renamed) -> "@floating-ui/react": "^0.26.13"
* "react-beautiful-dnd": "^13.1.0" (dead) -> "@hello-pangea/dnd": "^16.6.0" (fork)
* "focus-visible": "^5.2.0" -> removed, unnecessary
* "framer-motion": "^5.4.5" -> ^11.1.7
* "react": "^17.0.2" -> ^18.3.0 (cutting edge ok)
* "react-dom": "^17.0.2" -> ^18.3.0
* "react-joyride": "^2.4.0" -> ^2.8.1
* "react-markdown": "^8.0.3" -> ^9.0.1
* "remark-breaks": "^3.0.2" -> ^4.0.0 (required for compat with rmd 9)
* "remark-gfm": "^3.0.1" -> ^4.0.0 (required for compat with rmd 9)
* "type-fest": "^2.8.0" -> ^4.17.0 (synced across repo)
* "@testing-library/dom": "^8.11.1" -> removed, unnecessary
* "@testing-library/react": "^12.1.2" -> ^15.0.5 (required or tests will break)
* "@testing-library/user-event": "^14.4.3" -> ^14.5.2

refrained from upgrading some packages that are not super required, will do it in later PRs.
This includes:
@tanstack/react-query
react-hook-form, etc

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7292?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7308** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7305** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7305?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7292** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7292?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7285** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7285?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7279** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7279?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7278** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7278?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7277** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7277?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karrui** [comment]
almost there, all the regressions are hopefully caught by frontend tests. thanks @opengovsg/formsg-engineers for the upkeep of tests! it's coming in super handy now :)

**Comment by timotheeg** [comment]
Daaaaaang, great upgrades FTW @karrui 💪 !!

What noticeable improvements from the upgrade can the team expect, and should be monitored?

**Comment by karrui** [comment]
> Daaaaaang, great upgrades FTW @karrui 💪 !!
> 
> 
> 
> What noticeable improvements from the upgrade can the team expect, and should be monitored?

I think improvements are not noticeable to the public haha. This one just foundation to allow FormSG to use OGPDS (and not have upgrade path to react 19 get blocked when that comes out this year)


Developer experience wise, should be must faster to start up. Tests also run faster.

File: frontend/src/theme/components/Drawer.ts [hunk 1]
```diff
@@ -9,22 +10,31 @@ import { textStyles } from '~theme/textStyles'
...
 
+const baseStyleDialog = defineStyle({
+  [$bg.variable]: 'white',
+})
+
 const baseStyleHeader: SystemStyleObject = {
   ...textStyles['h2'],
   py: '1.25rem',
   px: '2.5rem',
+  bg: 'white',
```
Comment (karrui) [inline frontend/src/theme/components/Drawer.ts]: to remove, since dialog itself is white already

Comment (KenLSM) [reply frontend/src/theme/components/Drawer.ts]: [4c43914](https://github.com/opengovsg/FormSG/pull/7292/commits/4c43914fa37aea5570ca22296e85a96f1f21734b)

**Review by KenLSM — APPROVED** [review]

**Pull request merged**